drop-transfer/src/ws/events.rs

```diff
@@ -49,17 +53,25 @@ impl FileEventTxFactory {
 }
 
 impl<T: Transfer> FileEventTx<T> {
-    fn file_info(&self) -> Option<FileInfo> {
-        Some(self.xfer.files()[&self.file_id].info())
+    fn file_info(&self) -> FileInfo {
+        self.xfer.files()[&self.file_id].info()
     }
 
     async fn emit(&self, event: Event) {
-        let lock = self.inner.read().await;
+        let mut lock = self.inner.write().await;
```

Comment:
Now it's always writing. In that case, I think it should be changed to `tokio::sync::Mutex`